.github/scripts/test_and_publish.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+import subprocess, sys, os, tempfile
+from distantrs import Invocation
+from glob import glob
+
+INVOCATION_DATA_PATH = os.path.join(
+        tempfile.gettempdir(),
+        'distant-rs-invocation.txt'
+        )
+
+# required to find listener.py
+this_path = os.path.abspath(os.path.dirname(__file__))
+test_script_path = f"{os.environ.get('GITHUB_WORKSPACE')}/renode/test.sh"
+inv_data_str = ""
+robot_name = sys.argv[1].split(".")[0]
+
+try:
+    with open(INVOCATION_DATA_PATH, 'r') as f:
+        inv_data_str = f.read()
+except IOError:
+    print(str(e))
+    sys.exit(1)
+
+inv_data_lst = inv_data_str.split("--")
+
+i = Invocation(
+        invocation_id=inv_data_lst[0],
+        auth_token=inv_data_lst[1]
+        )
+
+i.announce_target(robot_name)
+
+tmp = tempfile.NamedTemporaryFile()
+with open(tmp.name, 'w') as log:
+    # run the actual testing procedure
+    subprocess_args = [
+            test_script_path, 
+            "--listener",
+            os.path.join(this_path, f"results_listener.py:{inv_data_str}"),
+            sys.argv[1],
+    ]
+
+    process = subprocess.Popen(
+        subprocess_args,
+        stdout=subprocess.PIPE, 
+        stderr=subprocess.STDOUT, 
+        env=os.environ.copy()
+        )
+
+    ret = 0
+
+    while True:
+        line = process.stdout.readline().decode('utf-8').rstrip()
+        ret = process.poll()
+        if ret is not None:
+            break
+        if line:
+            log.write(f'{line}\n')
+            print(line)
+
+
+i.add_log_to_target(
+    target_name=robot_name,
+    log_path=tmp.name
+    )
+
+dumps = glob('*.dump')
+
+expected_artifact_paths = [
+        'robot_output.xml',
+        'log.html',
+        'report.html',
+        ]
+expected_artifact_paths.extend(dumps)
```

Comment:
we could simplify the process if we extended this collection with pngs as well - just move artifacts handling below generating graphs

Reply:
Done. As a side effect, graphs will be generated even if tests fail.